internal/controllers/helpers.go

```
@@ -285,3 +298,49 @@ func createObject(ctx context.Context, c client.Client, obj client.Object) error
 
 	return nil
 }
+
+func getTrustedCAcert(ctx context.Context, cl client.Client, agentTLSModeFeatureEnabled bool) ([]byte, error) {
+	log := log.FromContext(ctx)
+
+	if !agentTLSModeFeatureEnabled {
+		log.Info("agent-tls-mode feature is disabled, using system store")
+		return nil, nil
+	}
+
+	agentTLSModeSetting := &managementv3.Setting{}
+
+	if err := cl.Get(ctx, client.ObjectKey{
+		Name: "agent-tls-mode",
+	}, agentTLSModeSetting); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("agent-tls-mode setting not found, using system store")
```

Nitpicking a bit. This does not align with Rancher, where the default is `strict` (since v2.9.0): https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/installation-references/tls-settings#agent-tls-enforcement. 

I would default to strict instead for this reason. I think it's also fair to just remove this branch and let it error out when this setting is not found, as this is not an expected scenario. The `agent-tls-mode` was added in Rancher v2.8